### PR TITLE
Implement svc handler and uart read

### PR DESCRIPTION
**Pull request by ryanpatiency**

### CODE REVIEW

File: 08-CMSIS/core/src/os.c [hunk 1]
```diff
@@ -5,29 +5,38 @@
...
 
 void test1(void *userdata)
 {
-	busy_loop(userdata);
+	int buf_size = 100;
+	char buf[buf_size];
+	while (1) {
+		puts("Enter command1: \r\n");
+		gets(buf, buf_size);
+		// delay(180000000);
```
Comment (jserv) [inline 08-CMSIS/core/src/os.c]: Don't add dead code.

File: 08-CMSIS/core/src/os.c [hunk 2]
```diff
@@ -5,29 +5,38 @@
...
 	while (count--) {
 	}
 }
 
 static void busy_loop(void *str)
 {
 	while (1) {
-		puts(str);
-		puts(": Running...\r\n");
+		// puts(str);
```
Comment (jserv) [inline 08-CMSIS/core/src/os.c]: Ditto. Remove dead code.

File: 08-CMSIS/Makefile [hunk 3]
```diff
@@ -43,11 +43,11 @@ $($(1)_TARGET):
 endef
 
 #Create corresponding build command to the device
-$(eval $(call eval_build_command,$(STM32P103_DEVICE)))
 $(eval $(call eval_build_command,$(STM32F429_DEVICE)))
+$(eval $(call eval_build_command,$(STM32P103_DEVICE)))
 
 #default target
-PLAT ?= $(STM32P103_DEVICE)
+PLAT ?= $(STM32F429_DEVICE)
```
Comment (jserv) [inline 08-CMSIS/Makefile]: Don't change the default target. Did you verify the changes on qemu-based stm32-p103 as well?

File: 08-CMSIS/Makefile [hunk 4]
```diff
@@ -59,6 +59,21 @@ target:
 st-flash: $($(STM32F429_DEVICE)_TARGET)
 	st-flash --reset write $($(STM32F429_DEVICE)_TARGET_bin) 0x8000000
 
+
+oocd:
+	sudo openocd -f board/stm32f429discovery.cfg  \
```
Comment (jserv) [inline 08-CMSIS/Makefile]: Don't hardcode board configurations here.

File: 08-CMSIS/core/src/itoa.c [hunk 5]
```diff
@@ -0,0 +1,20 @@
+void itoa(int i, char *a)
```
Comment (jserv) [inline 08-CMSIS/core/src/itoa.c]: Since `itoa` is only used by one C source, it is not worth to implement as public function.

**Comment by jserv** [comment]
Before merging the proposed changes, we have to ensure the following:
1. SVC handler has been verified on both QEMU-based stm32-p103 and stm32f429-discovery;
2. We are able to describe how SVC handler works and extend its usage;

**Review by lecopzer — CHANGES_REQUESTED** [review]
HI @ryanpatiency 
As @jserv said please verified does it work on all of platform.
Your commit need to rebase to clarify its function, I see some unrelated changes in a single commit,
and please don't expose merge conflict to pull request .

I think you have to explain EACH LINE why the change did in some critical path (like `svc_handler`), either in the commit message or comment in the source is okay. Every code has its meanings to exist, please remember this.

Thanks

**Comment by ryanpatiency** [comment]
Thank you for each detailed explanation, it is very helpful. I would work on the rebase ASAP.

File: 08-CMSIS/Makefile (hunk 4 above)
Comment (ryanpatiency) [reply 08-CMSIS/Makefile]: Why? isn't the `st-flash` above hardcode the board configuration?

Comment (jserv) [reply 08-CMSIS/Makefile]: No, we wish to support both QEMU-based stm32-p103 and stm32f429-discovery board.

**Pull request closed without merging**